plugin-infra/go-plugin-access/src/com/thoughtworks/go/plugin/access/notification/NotificationPluginRegistry.java

```diff
@@ -47,6 +47,10 @@ public void removePluginInterests(String pluginId) {
         }
     }
 
+    public boolean isAnyPluginInterestedIn(String notificationName) {
+        return !getPluginsInterestedIn(notificationName).isEmpty();
+    }
+
     public Set<String> getPluginsInterestedIn(String notificationName) {
```

Comment:
Write a throw-away test on this class, calling these methods in different threads (isAnyPluginInterestedIn and addToMap). I think there will be a concurrent modification exception around notificationNameToPluginsInterestedMap, or at least the Set<String> in it.